app/controllers/samples_reports_controller.rb

```diff
@@ -0,0 +1,133 @@
+class SamplesReportsController < ApplicationController
+  include SamplesReportsHelper
+
+  helper_method :boxes_data
+  helper_method :available_institutions
+
+  def index
+    @can_create = has_access?(@navigation_context.institution, CREATE_INSTITUTION_SAMPLES_REPORT)
+    @can_delete = has_access?(SamplesReport, DELETE_SAMPLES_REPORT)
+
+    @samples_reports = SamplesReport.where(institution: @navigation_context.institution)
+    @samples_reports = check_access(@samples_reports, READ_SAMPLES_REPORT).order('samples_reports.created_at DESC')
+  
+    # Filter by search params
+
+    @samples_reports = @samples_reports.partial_name(params[:name])
+    @samples_reports = @samples_reports.partial_sample_uuid(params[:sample_uuid])
+    @samples_reports = @samples_reports.partial_box_uuid(params[:box_uuid])
+    @samples_reports = @samples_reports.partial_batch_number(params[:batch_number])
+
+    #paginate samples report
+    @samples_reports = perform_pagination(@samples_reports)
+  end
+  
+  def new
+    @samples_report = SamplesReport.new({
+      institution: @navigation_context.institution,
+      site: @navigation_context.site
+    })
+
+    @boxes = []
+    
+    prepare_for_institution_and_authorize(@samples_report, CREATE_INSTITUTION_SAMPLES_REPORT)
+  end
+
+  def create
+    @institution = @navigation_context.institution
+    return unless authorize_resource(@institution, CREATE_INSTITUTION_SAMPLES_REPORT)
+
+    @samples_report = SamplesReport.new()
+    @samples_report.institution = @institution
+    @samples_report.site = @navigation_context.site
+    @samples_report.name = params[:samples_report][:name]
+    
+    samples_report_samples = []
+    @boxes = [] # For form reloading in case of validation error
+    if params[:samples_report][:boxes_attributes]
+      @boxes_p = params[:samples_report][:boxes_attributes]
+      @boxes_p.each do |box_param|
+        box_id = @boxes_p[box_param][:box_id]
+        box = Box.find_by_id(box_id)
+        box.samples.each do |sample|
+          samples_report_samples  << SamplesReportSample.new(samples_report: @samples_report, sample: sample)
+        end
+      end
+    end
+
+    @samples_report.samples_report_samples = samples_report_samples
+    
+    if @samples_report.save
+      redirect_to samples_reports_path, notice: 'Box report was successfully created.'
+    else
+      render action: 'new'
+    end
+  end
+
+  def show
+    @samples_report = SamplesReport.find_by_id(params[:id])
+    return unless authorize_resource(@samples_report, READ_SAMPLES_REPORT)
+    @can_delete = has_access?(@samples_report, DELETE_SAMPLES_REPORT)
+  end
+
+  def delete
+    @samples_report = SamplesReport.find_by_id(params[:id])
+    return unless authorize_resource(@samples_report, DELETE_SAMPLES_REPORT)
+  
+    SamplesReport.destroy(params[:id])
```

Comment:
Let's avoid reloading the SampleReport when we just did above :)
```suggestion
    @samples_report.destroy
```